src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,63 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.AddCommand;
+import duke.command.DeleteCommand;
+import duke.command.ListCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+
+import duke.task.Deadlines;
+import duke.task.Events;
+import duke.task.ToDos;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] splitCommands = fullCommand.split(" ", 2);
+        splitCommands[0] = splitCommands[0].strip().toUpperCase();
+        try {
+            switch (splitCommands[0]) {
+                case "TODO":
+                    ToDos todo = new ToDos(splitCommands[1]);
+                    return new AddCommand(todo);
+                case "DEADLINE":
+                    String[] splitDeadline = splitCommands[1].split("/by", 2);
+                    Deadlines deadline = new Deadlines(splitDeadline[0].strip(), LocalDate.parse(splitDeadline[1].strip()));
+                    return new AddCommand(deadline);
+                case "EVENT":
+                    String[] splitEvent = splitCommands[1].split("/at", 2);
+                    Events event = new Events(splitEvent[0].strip(), LocalDate.parse(splitEvent[1].strip()));
+                    return new AddCommand(event);
+                case "DELETE":
+                    return new DeleteCommand(Integer.parseInt(splitCommands[1].strip()));
+                case "LIST":
+                    return new ListCommand();
+                case "DONE":
+                    return new DoneCommand(Integer.parseInt(splitCommands[1].strip()));
+                case "BYE":
+                    return new ExitCommand();
+                default:
+                    throw new IllegalArgumentException();
+            }
```

Comment:
According to the coding style, case labels inside switch statements should not be indented further.